src/win32/posix_w32.c

```diff
@@ -55,7 +55,7 @@ int p_ftruncate(int fd, git_off_t size)
 		return -1;
 	}
 
-#if !defined(__MINGW32__)
+#if !defined(__MINGW32__) || defined(__MINGW64_VERSION_MAJOR)
```

Comment:
A recent PR suggests that [`MINGW_HAS_SECURE_API`](https://github.com/karvaporo/libgit2/commit/73682705093e028af823759b1ede5c9f1ec029fc) might be defined when this is supported.  This might be more idiomatic and future-proof.